internal/playlist_creator/FileData.go

```diff
@@ -0,0 +1,55 @@
+package playlist_creator
+
+import (
+	"fmt"
+	"playlistCreator/internal/playlist_creator/config"
+	"strings"
+)
+
+type FileEntry struct {
+	Path     string
+	FileName string
+}
+
+type FileData struct {
+	FilesList        []FileEntry
+	UniqueExtensions map[string]int
+}
+
+func (fileData *FileData) addFileExtension(extension string) {
+	var extensionLowercase = strings.ToLower(extension)
+	if fileData.UniqueExtensions == nil { //TODO: Can this initialisation be moved to a getter?
```

Comment:
yes, it can.

You can either use `sync.Once` to initialize. or have a `New` function that initializes for you.
eg:

```go
func New() *FileData {
return &FileData {
 uniqueExtensions: make(map[string]int)
}
}
```